pkg/plugin/v1/api.go

```diff
@@ -89,85 +82,43 @@ func (p createAPIPlugin) UpdateContext(ctx *plugin.Context) {
 	)
 }
 
-const (
-	group            = "group"
-	version          = "version"
-	kind             = "kind"
-	helmChart        = "helm-chart"
-	helmChartRepo    = "helm-chart-repo"
-	helmChartVerison = "helm-chart-version"
-
-	crdVersion        = "crd-version"
-	crdVersionV1      = "v1"
-	crdVersionV1beta1 = "v1beta1"
-)
-
+// BindFlags bind all plugin flags
 func (p *createAPIPlugin) BindFlags(fs *pflag.FlagSet) {
-	p.createOptions = chartutil.CreateOptions{}
+	p.apiFlags.AddTo(fs)
 	fs.SortFlags = false
-
-	fs.StringVar(&p.createOptions.GVK.Group, group, "", "resource Group")
-	fs.StringVar(&p.createOptions.GVK.Version, version, "", "resource Version")
-	fs.StringVar(&p.createOptions.GVK.Kind, kind, "", "resource Kind")
-
-	fs.StringVar(&p.createOptions.Chart, helmChart, "", "helm chart")
-	fs.StringVar(&p.createOptions.Repo, helmChartRepo, "", "helm chart repository")
-	fs.StringVar(&p.createOptions.Version, helmChartVerison, "", "helm chart version (default: latest)")
-
-	fs.StringVar(&p.createOptions.CRDVersion, crdVersion, crdVersionV1, "crd version to generate")
 }
 
+// InjectConfig will inject the PROJECT file/config in the plugin
 func (p *createAPIPlugin) InjectConfig(c *config.Config) {
 	p.config = c
 }
 
+// Run will call the plugin actions
 func (p *createAPIPlugin) Run() error {
 	return cmdutil.Run(p)
 }
 
+// Validate perform the required validations for this plugin
 func (p *createAPIPlugin) Validate() error {
-	if p.createOptions.CRDVersion != "v1" && p.createOptions.CRDVersion != "v1beta1" {
-		return fmt.Errorf("value of --%s must be either %q or %q", crdVersion, crdVersionV1, crdVersionV1beta1)
-	}
-
-	if len(strings.TrimSpace(p.createOptions.Chart)) == 0 {
-		if len(strings.TrimSpace(p.createOptions.Repo)) != 0 {
-			return fmt.Errorf("value of --%s can only be used with --%s", helmChartRepo, helmChart)
-		} else if len(p.createOptions.Version) != 0 {
-			return fmt.Errorf("value of --%s can only be used with --%s", helmChartVerison, helmChart)
-		}
+	// ensure that has config and the layout is valid for this plugin
+	if !hasPluginConfig(p.config) {
+		return fmt.Errorf("missing configuration for the helm plugin layout %v", plugin.KeyFor(Plugin{}))
 	}
```

Comment:
Note that the plugin create api not has the same validations of the init necessary. E.g ensure that the Project/config is scaffolded.